generator/Data/XCB/Python/Parse.hs

```diff
@@ -283,14 +285,50 @@ structElemToPyUnpack :: Expr ()
                      -> TypeInfoMap
                      -> GenStructElem Type
                      -> Either (Maybe String, String)
-                               (String, Expr (), Expr (), Maybe Int)
+                               (String, [(Expr (), Expr (), Maybe (Expr ()))], Maybe Int)
 structElemToPyUnpack _ _ _ (Pad i) = Left (Nothing, mkPad i)
 
 -- XXX: This is a cheap hack for noop, we should really do better.
 structElemToPyUnpack _ _ _ (Doc _ _ _) = Left (Nothing, "")
--- XXX: What does fd/switch mean? we should implement it correctly
+-- XXX: What does fd mean? we should implement it correctly
 structElemToPyUnpack _ _ _ (Fd _) = Left (Nothing, "")
-structElemToPyUnpack _ _ _ (Switch _ _ _) = Left (Nothing, "")
+
+-- The switch fields pick the way to expression to pack based on the expression
+structElemToPyUnpack unpacker ext m (Switch name expr bitcases) =
+  let cmp = xExpressionToPyExpr ((++) "self.") expr
+      switch = map (mkSwitch cmp) bitcases
+      switch' = map (pkSwitch unpacker ext m) switch
+  in Right (name, switch', Nothing)
+    where
+      mkSwitch :: Expr ()
+               -> BitCase
+               -> (GenStructElem Type, Expr ())
+      mkSwitch cmp (BitCase _ bc_cmp elems) =
+        let cmp_val = xExpressionToPyExpr id bc_cmp
```

Comment:
Haskellers mostly use camelCase instead of snake case, so it may be worth changing here.